### PR TITLE
IG-13146

### PR DESCRIPTION
**Pull request by imakhlin**
- [x] Fix conditional expression
- [x] Address PR comments
- [x] Refactor tests to support following data types: Int, Float, String
- [x] Update dependencies
- [x] Fix makefile

### CODE REVIEW

File: pkg/appender/ingest.go [hunk 1]
```diff
@@ -300,8 +301,17 @@ func (mc *MetricsCache) handleResponse(metric *MetricState, resp *v3io.Response,
 			// Metrics with too many update errors go into Error state
 			metric.retryCount++
 			if e, hasStatusCode := resp.Error.(v3ioerrors.ErrorWithStatusCode); hasStatusCode && e.StatusCode() != http.StatusServiceUnavailable {
-				mc.logger.ErrorWith(fmt.Sprintf("Chunk update failed with status code %d.", e.StatusCode()))
-				setError(mc, metric, errors.Wrap(resp.Error, fmt.Sprintf("Chunk update failed due to status code %d.", e.StatusCode())))
+				// If condition was evaluated as false log this and discard error.
```
Comment (talIguaz) [inline pkg/appender/ingest.go]: wrong comment

File: pkg/appender/ingest.go [hunk 2]
```diff
@@ -373,3 +383,22 @@ func setError(mc *MetricsCache, metric *MetricState, err error) {
...
 }
+
+const (
+	errorCodeString              = "ErrorCode"
+	falseConditionOuterErrorCode = "16777244"
+	falseConditionInnerErrorCode = "16777245"
+)
+
+// Check if the current error was caused specifically because the condition was evaluated to false.
+func isFalseConditionError(err error) bool {
```
Comment (talIguaz) [inline pkg/appender/ingest.go]: as i said yesterday, i also have this method for my delete PR. so let's put this peace of code in some common utility pkg

Comment (imakhlin) [reply pkg/appender/ingest.go]: np, moving to `utils/misc.go`

File: pkg/tsdb/v3iotsdb_integration_test.go [hunk 3]
```diff
@@ -45,6 +47,7 @@ import (
 const defaultStepMs = 5 * tsdbtest.MinuteInMillis // 5 minutes
 
 func TestIngestData(t *testing.T) {
+	timestamp := fmt.Sprintf("%d", time.Now().Unix()) //time.Now().Format(time.RFC3339)
```
Comment (talIguaz) [inline pkg/tsdb/v3iotsdb_integration_test.go]: delete comment

File: pkg/tsdb/v3iotsdb_integration_test.go [hunk 4]
```diff
@@ -158,13 +201,26 @@ func testIngestDataCase(t *testing.T, testParams tsdbtest.TestParams) {
...
 
 		if _, err := appender.WaitForCompletion(0); err != nil {
-			t.Fatalf("Failed to wait for appender completion. reason: %s", err)
+			if !isExpected(testParams, err) {
+				t.Fatalf("Failed to wait for appender completion. reason: %s", err)
+			}
 		}
 
-		tsdbtest.ValidateCountOfSamples(t, adapter, dp.Name, len(dp.Data), from, to, -1)
+		expectedCount := len(dp.Data)
```
Comment (talIguaz) [inline pkg/tsdb/v3iotsdb_integration_test.go]: i would prefer validating the values+count.
You can potentially add an `expectedData` rather then `expectedCount` and validate it against the actual data.
if `expectedData` is not set. expected will be the same as the "ingested" data.

File: pkg/chunkenc/vartype.go [hunk 5]
```diff
@@ -108,36 +108,34 @@ func (a *varAppender) Chunk() Chunk {
...
 		}
-
 		if math.IsNaN(val) {
 			a.appendNoValue(t, varTypeFloat64, varValueNone)
 			return
 		}
-
 		a.appendWithUint(t, varTypeFloat64, math.Float64bits(val))
-
+	case int:
```
[the remark this answers is not in the document]
Comment (imakhlin) [reply pkg/chunkenc/vartype.go]: this is not new code, I only added a case for `int` data type

File: pkg/tsdb/v3iotsdb_integration_test.go (hunk 4 above)
Comment (imakhlin) [reply pkg/tsdb/v3iotsdb_integration_test.go]: ok, so do it. I don't think that this PR should contain everything you miss in this project :-)

File: pkg/chunkenc/vartype.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (imakhlin) [reply pkg/chunkenc/vartype.go]: commented out, but not removed it yet. CI passes, so I can remove it finally if you wish

**Pull request merged**